Import TabletState and remove mutex in TabletMeta

chaoyli opened the pull request:
1.  Import TabletState to describe tablet state transitions.
2. There is no necessary to use std::mutex in TabletMeta,
     because concurrency conflict addressed by Tablet.

Commit:
Import TabletState and remove mutex in TabletMeta
1. Import TabletState to describe tablet state transitions.
2. There is no necessary to use std::mutex in TabletMeta,
   because concurrency conflict addressed by Tablet.

kangpinghuang commented on be/src/olap/tablet_manager.cpp:
```diff
@@ -555,8 +552,8 @@ TabletSharedPtr TabletManager::get_tablet(TTabletId tablet_id, SchemaHash schema
-            if (tablet->load() != OLAP_SUCCESS) {
+        } else if (load_tablet && !tablet->init_success()) {
+            if (tablet->init() != OLAP_SUCCESS) {
                 OLAP_LOG_WARNING("fail to load tablet. [tablet=%ld]", tablet_id);
```
LOG(INFO)

chaoyli replied:
OK

kangpinghuang commented on be/src/olap/tablet_manager.cpp:
```diff
@@ -555,8 +552,8 @@ TabletSharedPtr TabletManager::get_tablet(TTabletId tablet_id, SchemaHash schema
             tablet.reset();
-        } else if (load_tablet && !tablet->is_loaded()) {
-            if (tablet->load() != OLAP_SUCCESS) {
+        } else if (load_tablet && !tablet->init_success()) {
```
I think is_inited is a better name?

chaoyli replied:
I think init_succeeded may be better to describe it.
is_inited can not differentiate success or fail literally

kangpinghuang commented on be/src/olap/tablet.h:
```diff
@@ -27,221 +27,211 @@
-    void set_id(size_t id) { _id = id; }
+    inline DataDir* data_dir() const;
     OLAPStatus register_tablet_into_dir();
+    std::string dir_path() const;
```
I think dir_path is not a good name? data_dir_path?

chaoyli replied:
OK, I remove dir_path() from tablet.h

kangpinghuang commented on be/src/olap/tablet.cpp:
```diff
@@ -73,8 +56,8 @@ TabletSharedPtr Tablet::create_from_tablet_meta_file(
-    std::string shard_path = file_path_path.parent_path().parent_path().parent_path().string();
-    std::string shard_str = shard_path.substr(shard_path.find_last_of('/') + 1);
+    string shard_path = file_path_path.parent_path().parent_path().parent_path().string();
+    string shard_str = shard_path.substr(shard_path.find_last_of('/') + 1);
```
you delete using std::string, can this be compiled?

chaoyli replied:
using std::string in tablet.h

kangpinghuang commented on be/src/olap/base_compaction.cpp:
```diff
@@ -200,7 +200,7 @@ bool BaseCompaction::_check_whether_satisfy_policy(bool is_manual_trigger,
         if (temp.first == 0) {
             _old_base_version = temp;
-            base_size = _tablet->get_version_data_size(temp);
+            base_size = _tablet->get_rowset_size_by_version(temp);
```
temp is not a good name?

kangpinghuang commented on be/src/olap/base_compaction.cpp:
```diff
@@ -43,7 +43,7 @@ namespace doris {
     // 表在首次查询或PUSH等操作时，会被加载到内存
     // 如果表没有被加载，表明该表上目前没有任何操作，所以不进行BE操作
-    if (!tablet->is_loaded()) {
+    if (!tablet->init_success()) {
```
init_success -> is_inited?

chaoyli commented on be/src/olap/tablet_meta.h:
```diff
@@ -91,29 +112,51 @@ class TabletMeta {
 
-    OLAPStatus to_tablet_pb(TabletMetaPB* tablet_meta_pb);
-    OLAPStatus to_tablet_pb_unlock(TabletMetaPB* tablet_meta_pb);
+    inline void set_data_dir(DataDir* data_dir);
```
When cloning tablet_meta from remote host, data_dir is empty.

chaoyli commented on be/src/olap/tablet_meta.h:
```diff
@@ -123,80 +166,55 @@ class TabletMeta {
     DataDir* _data_dir;
-
-    mutable std::mutex _mutex;
+    TabletMetaPB _tablet_meta_pb;
```
In construtor

Commit:
Import TabletState and remove mutex in TabletMeta
1. Import TabletState to describe tablet state transitions.
2. There is no necessary to use std::mutex in TabletMeta,
   because concurrency conflict addressed by Tablet.
